src/components/UserInfo/UserInfoCard/index.tsx

```diff
@@ -0,0 +1,228 @@
+import { useCallback } from 'react';
+import { MAXEXP } from '@/common/number';
+import * as S from './styles';
+import {
+  UserInfoMockList as userList,
+  userQuizMockList as userQuizList,
+  UserInfoMockData as userMockData,
+} from '@/assets/UserInfoMockData';
+import * as Breakpoints from './breakpoints';
+
+interface userProps {
+  id: string;
+}
+
+function UserInfoCard({ id }: userProps) {
+  // TODO: id를 기반으로 비동기 API요청> userData 가져오기
+  const userData = {
+    id: userMockData._id,
+    fullName: userMockData.fullName,
+    likes: userMockData.likes,
+    posts: userMockData.posts,
+    comments: userMockData.comments,
+    totalExp: JSON.parse(userMockData.username).totalPoints,
+  };
+
+  // TODO: 유저리스트 비동기 API요청 필요
+  const getRank = useCallback(() => {
+    const sortedUserList = userList
+      .map((user) => ({
+        id: user._id,
+        fullName: user.fullName,
+        points: JSON.parse(user.username).totalPoints,
+      }))
+      .sort((a, b) => {
+        return b.points - a.points;
+      });
+    const rank = sortedUserList.findIndex((data) => data.id === id) + 1;
+    return rank;
+  }, [id]);
+
+  // TODO: 유저가 작성한 퀴즈 정보 비동기 API요청 필요
+  const getQuiz = () => {
+    const modifiedQuiz = userQuizList.map((quiz) => {
+      const customData = JSON.parse(quiz.title);
+      return {
+        id: quiz._id,
+        category: customData.tag,
+      };
+    });
+    return modifiedQuiz;
+  };
+
+  const level = Math.floor(userData.totalExp / MAXEXP) + 1;
+  const currentExp = userData.totalExp - (level - 1) * MAXEXP;
+  const expPercent = Math.floor((currentExp / MAXEXP) * 100);
+
+  const getImage = useCallback(() => {
+    let selectedId = Breakpoints.imageBreakpoints[0].imageId;
+    Breakpoints.imageBreakpoints.forEach((breakpoint) => {
+      if (level >= breakpoint.level) {
+        selectedId = breakpoint.imageId;
+      }
+    });
+    return selectedId;
+  }, [level]);
+
+  interface badgeType {
+    id: string;
+    color?: string;
+    content: string;
+  }
+
+  const getBadges = () => {
+    const badges: badgeType[] = [];
+
+    // 레벨 별 뱃지
+    let levelBadgeId = 0;
+    Breakpoints.levelBreakpoints.forEach((badge, index) => {
+      if (level > badge.level) {
+        levelBadgeId = index;
+      }
+    });
```

Comment:
사실 levelBadgeId는 해당 breakpoints의 idx를 나타내고 있으니 selectedBadgeIdx 가 더 적합하지 않을까 싶네요 ㅎㅎ
저는 let과 forEach 사용보다, 고차함수 등을 활용한 방식을 지향하고 있어서, 생각해보긴했는데, 코드가 지저분 해보이긴 하네요.
아마 조금 더 생각하면 좀 더깔끔히 나올듯도 싶은데, 그냥 의견남깁니다 ㅎㅎ.
```suggestion
    const selectedBadge = Breakpoints.levelBreakpoints.filter(
      (badge, idx, bgList) => {
        const isLastBadge = idx === bgList.length - 1;
        return isLastBadge
          ? badge
          : level > badge.level && level <= bgList[idx + 1].level;
      },
    );
```

Reply:
의견 남겨주셔서 감사합니다! 저도 이 부분 작성하면서 let 지양하는 쪽으로 구현하고 싶다고 생각했는데 잘 생각이 안나더라구요.. 인수님 의견 좋은 것 같아요! 이대로나, 아니면 조금 더 생각해서 바꿔보는 방향으로 하겠습니다 😄